x-pack/elastic-agent/pkg/agent/control/server/listener.go

```diff
@@ -48,7 +45,7 @@ func createListener(log *logger.Logger) (net.Listener, error) {
 
 func cleanupListener(log *logger.Logger) {
 	path := strings.TrimPrefix(control.Address(), "unix://")
-	if err := os.Remove(path); err != nil {
-		log.Errorf("%s", errors.New(err, fmt.Sprintf("Failed to cleanup %s", path), errors.TypeFilesystem, errors.M("path", path)))
+	if err := os.Remove(path); err != nil && !os.IsNotExist(err) {
+		log.Debug("%s", errors.New(err, fmt.Sprintf("Failed to cleanup %s", path), errors.TypeFilesystem, errors.M("path", path)))
```

Comment:
In which scenarios does the file not exists so the cleanup fails?

Reply:
I believe the case is when the temp directory is removed before the actually socket. So that cause it to fail on the removal.

Reply:
But what removes the temp directory? BTW, don't want to block this PR on this but would be good to understand why it happens.